### PR TITLE
appservice: Refactor deployment code into steps

### PR DESCRIPTION
**Pull request by nturinski**

### CODE REVIEW

File: appservice/src/deploy/deploy.ts [hunk 1]
```diff
@@ -3,121 +3,24 @@
...
-        }
-    });
-}
-
-function getLinuxFxVersionForTelemetry(config: SiteConfigResource): string {
-    const linuxFxVersion = config.linuxFxVersion || '';
-    // Docker values point to the user's specific image, which we don't want to track
-    return /^docker/i.test(linuxFxVersion) ? 'docker' : linuxFxVersion;
+    const innerContext: InnerDeployContext = Object.assign(context, { site, fsPath, client, aspPromise });
+    const title: string = l10n.t('Deploying to function app "{0}"', site.fullName);
```
Comment (alexweininger) [inline appservice/src/deploy/deploy.ts:21]: Function is used by App Service and Functions so title needs to either be generic or handle both cases.

```suggestion
    const title: string = l10n.t('Deploying to "{0}"', site.fullName);
```

File: appservice/src/deploy/runWithZipStream.ts [hunk 2]
```diff
@@ -12,20 +12,23 @@ import * as prettybytes from 'pretty-bytes';
...
     site: ParsedSite,
     pathFileMap?: Map<string, string>
+    progress?: vscode.Progress<{ message?: string; increment?: number }>
     callback: (zipStream: Readable) => Promise<AzExtPipelineResponse | void>
 }): Promise<AzExtPipelineResponse | void> {
 
     function onFileSize(size: number): void {
         context.telemetry.measurements.zipFileSize = size;
+        const zipFileSize = vscode.l10n.t('Zip package size: {0}', prettybytes(size));
         ext.outputChannel.appendLog(vscode.l10n.t('Zip package size: {0}', prettybytes(size)), { resourceName: site.fullName });
```
Comment (alexweininger) [inline appservice/src/deploy/runWithZipStream.ts:30]:
```suggestion
        ext.outputChannel.appendLog(zipFileSize, { resourceName: site.fullName });
```

File: appservice/src/deploy/wizard/DeployExecuteBaseStep.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AppServicePlan, SiteConfigResource } from "@azure/arm-appservice";
+import { ext } from "@microsoft/vscode-azext-github";
+import { AzureWizardExecuteStep, randomUtils } from "@microsoft/vscode-azext-utils";
+import { Progress, ProgressLocation, l10n, window } from "vscode";
+import { InnerDeployContext } from "../IDeployContext";
+
+export abstract class DeployExecuteBaseStep extends AzureWizardExecuteStep<InnerDeployContext> {
```
Comment (alexweininger) [inline appservice/src/deploy/wizard/DeployExecuteBaseStep.ts:12]: `DeployExecuteStepBase` might be a better name.

**Comment by alexweininger** [comment]
Overall looks really good. I tested it myself with a few simple Function App deploys.

**Review by alexweininger — DISMISSED** [review]
Fix da linter

**Comment by nturinski** [comment]
Bump 👊

**Comment by nturinski** [comment]
I think this is ready to review/approve btw.

**Review by alexweininger — APPROVED** [review]
I approve, however lets properly plan out when we'll actually release this. I don't think we want to include this in the upcoming releases.

**Comment by JasonYeMSFT** [comment]
FYI, the zip deployment code has been modified on main. I don't know why there isn't any merge conflicts.

**Comment by nturinski** [comment]
Probably because it's all in new files. I'll be sure to transfer the changes over.

**Comment by nturinski** [comment]
I think everything is okay with this one. Gonna go ahead and just merge it with Alex's dismissed approval

**Pull request merged**